pinot-controller/src/main/resources/webapp/js/init.js

```diff
@@ -98,6 +98,7 @@ var HELPERS = {
 
   executeQuery: function(query, traceEnabled, callback) {
     var url = "/pql?pql=" + query +"&trace=" + traceEnabled;
+    url = encodeURI(url);
```

Comment:
Oh, interesting, `encodeURI` doesn't actually encode special characters like & and ?. In your case, it worked fine, but it means that if there is a query that contains a question mark, the question mark won't get encoded properly.

What you want in this case is `encodeURIComponent(query)` which will properly escape the query contents:

```
encodeURI('foo=bar?baz%')
"foo=bar?baz%25"
encodeURIComponent('foo=bar?baz%')
"foo%3Dbar%3Fbaz%25"
```